### PR TITLE
Remove ahead-of-time compilation (AOT)

### PR DESCRIPTION
**Pull request by adk9**
This PR removes the now-deprecated ahead-of-time compilation feature. Tools and utilities, such as `tf2ngraph`, that were relevant for AoT have also been removed.

### CODE REVIEW

File: ngraph_bridge/ngraph_encapsulate_impl.h [hunk 1]
```diff
@@ -44,17 +44,12 @@ class NGraphEncapsulateImpl {
                           std::vector<const Tensor*>& static_input_map,
                           std::stringstream& signature_ss);
 
-  static Status Compile(std::shared_ptr<ngraph::Function> ng_function,
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_encapsulate_impl.h:47]: Are these functions specific to AOT?

File: ngraph_bridge/grappler/ngraph_optimizer.cc [hunk 2]
```diff
@@ -208,9 +180,7 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
 
   // 4. Encapsulate clusters then, if requested, dump the graphs.
   FunctionDefLibrary* fdeflib_new = new FunctionDefLibrary();
-  // TODO: right now _ngraph_aot_requested is passed along in config_map.
```
Comment (kanvi-nervana) [inline ngraph_bridge/grappler/ngraph_optimizer.cc:211]: Is the FunctionDefLibrary still needed?

File: ngraph_bridge/ngraph_encapsulate_clusters.h [hunk 3]
```diff
@@ -49,8 +42,7 @@ typedef std::pair<bool, std::set<ShapeHintMap>> AOTInfo;
 /// compilation.
```
[the remark this answers is not in the document]
Comment (adk9) [reply ngraph_bridge/ngraph_encapsulate_clusters.h]: Fixed.

File: ngraph_bridge/grappler/ngraph_optimizer.cc (hunk 2 above)
Comment (adk9) [reply ngraph_bridge/grappler/ngraph_optimizer.cc:211]: No, removed.

File: ngraph_bridge/ngraph_encapsulate_impl.h (hunk 1 above)
Comment (adk9) [reply ngraph_bridge/ngraph_encapsulate_impl.h:47]: No, but they were not needed, hence removed.

**Review by bani-intelaipg — APPROVED** [review]

**Pull request merged**